internal/regopolicyinterpreter/regopolicyinterpreter.go

```diff
@@ -329,15 +361,8 @@ func (r *RegoPolicyInterpreter) DisableLogging() error {
 	return nil
 }
 
-// Compile compiles the policy and its modules. This will increase the speed of policy
-// execution.
-func (r *RegoPolicyInterpreter) Compile() error {
-	r.mutex.Lock()
-	defer r.mutex.Unlock()
-
-	if r.compiledModules != nil {
-		return nil
-	}
+func (r *RegoPolicyInterpreter) compile() error {
+	// dataAndModulesMutex must be held before calling this
```

Comment:
the comment should ideally go to the function docstring.

Reply:
I'll add this to the nit list.